octobot_trading/exchanges/exchange_builder.py

```diff
@@ -0,0 +1,117 @@
+
+        self.exchange_manager: ExchangeManager = ExchangeManager(self.config, self.exchange_name)
+
+        self._is_using_trading_modes: bool = True
+        self._is_exchange_manager_sandboxed: bool = False
+
+        self._trading_tentacles_path = CONFIG_TRADING_FILE_PATH
+
+    """
+    Build
+    """
+    async def build(self):
+        await self.exchange_manager.initialize()
+
+        # initialize trader
+        if self.exchange_manager.trader is not None:
+            await self._build_trader()
+
+        # create trading modes
+        if self._is_using_trading_modes:
+            await self._build_modes()
+
+        # configure exchange
+        if not self.exchange_manager.is_backtesting:
+            self.exchange_manager.exchange.client.setSandboxMode(self._is_exchange_manager_sandboxed)
+
+        # add to global exchanges
+        Exchanges.instance().add_exchange(self.exchange_manager)
+
+        return self.exchange_manager
+
+    async def _build_trader(self):
+        try:
+            # check traders activation
+            if not is_trader_enabled(self.config) and not is_trader_simulator_enabled(self.config):
+                raise ValueError(f"No trader simulator nor real trader activated on "
+                                 f"{self.exchange_manager.exchange_name}")
+
+            await self.exchange_manager.trader.initialize()
+        except ValueError as e:
+            self.logger.error(f"An error occurred when creating trader : ")
+            raise e
+
+    async def _build_modes(self):
+        try:
+            init_trading_mode_config(self.config, self._trading_tentacles_path)
+            await create_trading_mode(self.config, self.exchange_manager)
+        except Exception as e:
+            self.logger.error(f"An error occurred when initializing trading mode : ")
+            raise e
+
+    """
+    Builder methods
+    """
+    def is_backtesting(self, backtesting_files):
+        self.exchange_manager.is_backtesting = True
+        self.exchange_manager.backtesting_files = backtesting_files
+        return self
+
+    def is_sandboxed(self, sandboxed: bool):
+        self._is_exchange_manager_sandboxed = sandboxed
+        return self
+
+    def is_simulated(self):
+        self.exchange_manager.is_simulated = True
+        self.exchange_manager.trader = TraderSimulator(self.config, self.exchange_manager)
+        return self
+
+    def is_real(self):
+        self.exchange_manager.is_simulated = False
+        self.exchange_manager.trader = Trader(self.config, self.exchange_manager)
+        return self
+
+    def is_rest_only(self):
+        self.exchange_manager.rest_only = True
+        return self
+
+    def use_trading_mode(self, trading_tentacles_path):
+        self._is_using_trading_modes = True
+        self._trading_tentacles_path = trading_tentacles_path
+        return self
+
+    def disable_trading_mode(self):
+        self._is_using_trading_modes = False
+        return self
```

Comment:
Can you add an self.exchange_manager.ignore_config setter plz ?

Reply:
Can you also add a seld.exchange_manager.exchange_only plz ? :)

Reply:
There is already a ` is_rest_only()`, isn't it what you need ?